geostore/s3_utils.py

```diff
@@ -11,12 +11,20 @@ def get_bucket_and_key_from_url(url: str) -> Tuple[str, str]:
     return parsed.netloc, parsed.path[1:]
 
 
-def get_s3_url_reader(s3_role_arn: str) -> Callable[[str], StreamingBody]:
-    def s3_url_reader(url: str) -> StreamingBody:
+class GeostoreS3Response:
+    # pylint: disable=too-few-public-methods
+    def __init__(self, response: StreamingBody):
```

Comment:
The `StreamingBody` is *part* of the S3 response, so it might be good to give it a different name. Maybe something like `body`, `stream`, `object_stream` or something like that.